### ISSUE TITLE
Duplicate column names when joining two whole tables

### ISSUE THREAD

**Issue by aljazerzen** [PRQL/prql#3133]
### What's up?

Follow up for 

Imagine we have two tables:
```elm
let artists <[{artist_id, name}]>
let albums <[{album_id, name, artist_id}]>
```

If you do this:
```elm
from artists
join albums (albums.artist_id == artists.artist_id)
```

Our semantics dictate that the following relation should be produced:
```
[{artists = {artist_id, name}, albums = {album_id, name, artist_id}}]
```
when this is compiled to SQL, nested tuples are flattened:

```sql
SELECT artist_id, name, album_id, name, artist_id
FROM artists
JOIN albums ON albums.artist_id = artists.artist_id
```
... which is problematic because:
- you can no longer differentiate between the two `artist_id`s (which are not necessarily exactly the same, because of NULLs)
- some dialects will report errors as reported in https://github.com/PRQL/prql/issues/879#issuecomment-1652401282

We can solve this quite easily, by instead compiling to:

```sql
SELECT "artists.artist_id", "artists.name", "albums.album_id", "albums.name", "albums.artist_id"
FROM artists
JOIN albums ON albums.artist_id = artists.artist_id
```
(... which would actually be more correct than what we currently have.)

The problem here is unknown tables. If the compiler has no knowledge of all the table columns, it produces this:
```sql
SELECT artists.*, artists.*
FROM artists
JOIN albums ON albums.artist_id = artists.artist_id
```
And even if `SELECT artists.* EXCLUDE (artist_id)`, there will be clashes on `name`.

TLDR; when we are joining two tables without complete knowledge of all the columns, it is not possible to produce valid SQL for some dialects

**Comment by max-sixty** [comment]
This is a big advantage of `USING`...

**Comment by richb-hanover** [comment]
I have a sense that this may be related to #2965...

**Comment by aljazerzen** [comment]
> This is a big advantage of USING...

Yes, it solves a part of the problem, but there still may be clashes between columns not in the USING clause.

**Comment by PrettyWood** [comment]
I would like to support nested joins but it fails

```prql
from `artists`
derive {
`agg` = f"{`name`} === {`artist_id`}"
}
join side:inner (
  from artists
    join side:inner (
    from `artists`
    derive {
      `concatenated` = f"{`name`} -> {`artist_id`}"
    }
  ) (this.artist_id==that.artist_id)
) (this.artist_id == that.artist_id)
```

And I reckon it's due to those duplicated columns.
And I don't seem to be able to make it work with `select {!...}`

I know I can unnest it myself but those PRQL queries are generated on the fly with substep so I don't have the full context

What workaround would you have in mind? Or can I work on a fix? If yes which one?
